app/src/main/java/com/eveningoutpost/dexdrip/LibreReceiver.java

```diff
@@ -137,9 +139,15 @@ public void run() {
                                 Log.v(TAG, "got bg reading: from sensor:" + currentRawValue.serial + " rawValue:" + currentRawValue.glucose + " at:" + currentRawValue.timestamp);
                                 // period of 4.5 minutes to collect 5 readings
                                 if (!BgReading.last_within_millis(DexCollectionType.getCurrentDeduplicationPeriod())) {
-                                    List<Libre2RawValue> smoothingValues = Libre2RawValue.last20Minutes();
+                                    long smoothing_minutes = LibreReceiver.libreFilterLength(xdrip.getAppContext());
```

Comment:
You should replace with `Pref.getStringToInt("libre_filter_length", 25)` or do the same in `libreFilterLength()` if you expect to need to reuse it. There is no need to manage shared preference and context yourself as this is already encapsulated inside the `Pref` helper class.